examples/generic-azure.yaml

```diff
@@ -672,4 +672,39 @@ groups:
 
           ## References
           - section 9.1.2 https://documentation.suse.com/sbp/all/single-html/SLES4SAP-hana-sr-guide-PerfOpt-15/
+      - id: 1.5.3
+        description: "ntp clock synchronization is working and using the slew mode"
+        audit: |
+          # The third field of ntpq -p is the stratum and if it's 16
+          # the sync is disabled. Make sure it's not 16
+          if ! sudo ntpq -pn | grep "No association ID's returned" \
+          && sudo ntpq -pn | awk '{print $3}' | grep -e "[0-9]" | grep -v 16; then
+            if ps aux | grep "/usr/sbin/ntpd" | grep "\-x"; then echo passed; fi
```

Comment:
this is quite a weak way of checking for this, as you're listing all processes (so including the grep process). the easiest fix is to ensure that grep doesn't find itself by making a regexp that doesn't match itself, e.g.

`grep "/usr/sbin/[n]tpd`

but you could also do a `grep -q --- -x /proc/$(</run/ntpd.pid)/cmdline` which would do pretty much the same thing..

Reply:
Marking this as resolved as it was already addressed